### PR TITLE
fix conditions in `lib/name`

### PR DESCRIPTION
**Pull request by bekzod**

### CODE REVIEW

File: packages/ember-metal/lib/namespace_search.js [hunk 1]
```diff
@@ -169,6 +169,7 @@ function tryIsNamespace(lookup, prop) {
     );
   } catch (e) {
     // continue
+    return false;
```
Comment (bekzod) [inline packages/ember-metal/lib/namespace_search.js]: so `tryIsNamespace` returns only `boolean`

Comment (rwjblue) [reply packages/ember-metal/lib/namespace_search.js]: Hmm, seems like the return inside the `try` block returns an object, not boolean.  Am I misunderstanding something?

Comment (bekzod) [reply packages/ember-metal/lib/namespace_search.js]: my bad sorry

**Review by krisselden — APPROVED** [review]

**Comment by rwjblue** [comment]
Thank you!

**Pull request merged**